app/main/presenters/search_presenters.py

```diff
@@ -288,8 +288,18 @@ def _annotate_categories_with_selection(lot_slug, category_filters, request, url
             content_manifest, framework, aggregations, keys_to_remove, parent_category=category['value'])
 
         if selected_descendants:
-            # If a parentCategory has been sent as a url query param, de-select all other parent categories.
-            if request.values.get('parentCategory', category['value']) != category['value']:
+            # If a parentCategory has been sent as a url query param, and it's this category...
+            # (Note: `get` fallback is for if there are selected descendants but no parent category is set, which is
+            # not expected, but could happen if constructing URLs by hand.)
+            if request.values.get('parentCategory', category['value']) == category['value']:
```

Comment:
I'm *really* confused over werkzeug's differentiation between `args` `form` and `values` and it seems from a distance to be one of the worst designed things ever.

Reply:
Me too. So I just looked it up: args is from the GET query string; form is from the POST; values is a combination of the two, e.g. for if your form might be a POST or a GET and you don't much know or care.

I suppose in these presenters, `values` is the right thing, because they don't know whether the view that called them was a POST or a GET. Really, they probably shouldn't be poking around in the request, so the selected parent category should have been passed in. But this was easier at the time, and now doesn't seem like the moment to revisit whether our search presenters are sufficiently decoupled...

Reply:
Yeah, the thing I can't figure out is what you're supposed to do if you want to access the POSTed parameters unsullied by any query strings. Anyway, not that relevant here...

Reply:
That's `request.form` - I think. In other words, that's not an actual form object (because that would introduce a dependency on wtforms or whatever, and flask doesn't do that) - it's just another multidict.

Reply:
Well I would think that if the docs didn't do on to say

```
Changed in version 0.9: Previous to Werkzeug 0.9 this would only contain form data for POST and PUT requests.
```

Which I take to mean pulls in the parameters from GET requests...?

Reply:
For the record, we've established it won't look in the URL/GET params, but regardless of HTTP method it will look in the request body, as long as the content type is `multipart/form-data` or equivalent.

Reply:
oh and this is probably clearer if written:

```
if 'parentCategory' not in request.values or request.values.get('parentCategory') == category['value']:
```

Reply:
You're right that the actual logic becomes more clear, but it calls out the weird case even more, which doesn't make the overall 'normal' flow more clear? Also I'd have to rewrite the comment, which already made my head hurt enough...

Reply:
Weird cases are just as important as the "normal" case to me but whatevs